unit_tests/common/range_tools.hpp

```diff
@@ -11,9 +11,11 @@
         const auto &rhs = {__VA_ARGS__};                                                           \
         BOOST_CHECK_EQUAL_COLLECTIONS(lhs.begin(), lhs.end(), rhs.begin(), rhs.end());             \
     } while (0)
-#define CHECK_EQUAL_COLLECTIONS(lhs, rhs)                                                          \
```

Comment:
Failed if `lhs` or `rhs` is expression creating a value (e.g. a literal) - `begin()` and `end()` would reference different containers.